nilearn/image/image.py

```diff
@@ -376,6 +394,22 @@ def crop_img(img, rtol=1e-8, copy=True, pad=True, return_offset=False):
         *[(x1_pre, x1_post), (x2_pre, x2_post), ..., (xN_pre, xN_post)]*
 
     """
+    # TODO: remove this warning in 0.13.0
+    if copy_header is None:
+        copy_header_default = (
+            "From release 0.13.0 onwards, this function will, by default, "
+            "copy the header of the input image to the cropped image."
+            "Currently, the header is reset to the default Nifti1Header."
+            "To suppress this warning and continue using the current behavior "
+            "set `copy_header=False`. To use the new behavior, set "
+            "`copy_header=True`."
+        )
+        warnings.warn(
+            category=FutureWarning,
+            message=copy_header_default,
+        )
+        copy_header = False
```

Comment:
That comment goes for the other functions as well.

Reply:
if you find it easier, you can also create a single utility function that checks the value of `copy_header` and throws a warning.

If you do may put it in thre : `nilearn\_utils\helpers.py` where there are a few other deprecation related utilities.

Also thinking that it may make it easier when future us (or future maintainers) have to deal with this deprecation.

Reply:
Indeed, I was wondering if we already have a system to track such deprecations.

In addition to the hassle of removing every single warning message and replacing every parameter later on, it is possible that we might just miss/forget about some deprecations when the said release is made.